test/tests/functional/pbs_soft_walltime.py

```diff
@@ -687,3 +709,336 @@ def test_preempt_order(self):
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 5}, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_soft_modifyjob_hook(self):
+        """
+        Test that soft walltime is set by modifyjob hook
+        """
+
+        hook_body = \
+            """import pbs
+e = pbs.event()
+e.job.Resource_List["soft_walltime"] = 15
+e.accept()
+"""
+        a = {'event': 'modifyjob', 'enabled': 'True'}
+        self.server.create_import_hook("que", a, hook_body)
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+
+        self.server.expect(
+            JOB, 'Resource_List.soft_walltime', op=UNSET, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 5})
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid)
+
+    def test_walltime_default(self):
+        """
+        Test soft walltime behavior with hard walltime is same
+        even if set under resource_default
+        """
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'Resources_default.soft_walltime': '15'})
+
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': 15})
+        jid = self.server.submit(J)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.deljob(jid, wait=True)
+
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': 16})
+        jid1 = self.server.submit(J)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 15}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.deljob(jid1, wait=True)
+
+        try:
+            J = Job(TEST_USER, attrs={'Resource_List.walltime': 10})
+            jid1 = self.server.submit(J)
+        except PtlSubmitError as e:
+            self.assertTrue("illegal resource" in e.msg[0])
+
+    def test_soft_suspension(self):
+        """
+        Test that if job is suspended soft_walltime will not get extended
+        """
+
+        J = Job(TEST_USER, {'Resource_List.walltime': 300})
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 7})
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        self.server.sigjob(jid, signal='suspend')
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+        time.sleep(7)
+        self.server.expect(JOB, {'Resource_List.soft_walltime': 7}, id=jid)
+
+        self.server.sigjob(jid, signal='resume')
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        time.sleep(5)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'estimated.soft_walltime':
+                                 (MATCH_RE, '00:00:14|14')}, id=jid)
```

Comment:
Hmm.. I see. I deliberately sleeping for 5s to avoid missing the window for estimated.soft_walltime=14, so that even if expect gets delayed by a sec or 2 will still catch it. I also ran into the race condition where expect and sleep took longer and soft_walltime failed.

Reply:
I'm starting to regret not fixing PP-1052 in PP-482's pull request.  I just found it so late in the game that I didn't.  Now it's making writing tests harder.  Why don't we leave this test alone until PP-1052 is fixed.  PP-1052 will probably require a few tests to be rewritten.

Reply:
or on another thought I added this test to verify that soft_walltime is not extended if job is suspended (or held or queued or waiting). I think I should still modify the test to test that objective and like you said we can add tests for PP-1052 later as part of bug fix.